### PR TITLE
fix(engine): mark replica as ERROR if it is not found

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/9166

#### What this PR does / why we need it:

Mark replica as ERROR if it is not found. Then, longhorn-manager will be aware of the failed replica and trigger cleanup.

### CODE REVIEW

File: pkg/spdk/engine.go [hunk 1]
```diff
@@ -648,7 +648,12 @@ func (e *Engine) checkAndUpdateInfoFromReplicaNoLock() {
 		}
 		replica, err := replicaServiceCli.ReplicaGet(replicaName)
 		if err != nil {
-			e.log.WithError(err).Warnf("failed to get replica %s with address %s, will skip this replica and continue info update from replica", replicaName, address)
+			if strings.Contains(err.Error(), "cannot find replica") {
+				e.log.WithError(err).Warnf("Failed to get replica %s with address %s, make it as ERR", replicaName, address)
+				e.ReplicaModeMap[replicaName] = types.ModeERR
+			} else {
+				e.log.WithError(err).Warnf("failed to get replica %s with address %s, will skip this replica and continue info update from replica", replicaName, address)
```
Comment (shuo-wu) [inline pkg/spdk/engine.go:655]: NIT: Correct the printed error log format then it's good to go

**Review by shuo-wu — DISMISSED** [review]

File: pkg/spdk/engine.go (hunk 1 above)
Comment (derekbit) [reply pkg/spdk/engine.go]: Fixed. Thank you.

**Comment by derekbit** [comment]
@shuo-wu There are some changes. Could you review it again? Thank you.

File: pkg/spdk/engine.go [hunk 2]
```diff
@@ -648,7 +648,8 @@ func (e *Engine) checkAndUpdateInfoFromReplicaNoLock() {
 		}
 		replica, err := replicaServiceCli.ReplicaGet(replicaName)
 		if err != nil {
-			e.log.WithError(err).Warnf("failed to get replica %s with address %s, will skip this replica and continue info update from replica", replicaName, address)
+			e.log.WithError(err).Warnf("Failed to get replica %s with address %s, make it as ERR", replicaName, address)
```
Comment (shuo-wu) [inline pkg/spdk/engine.go:651]: NIT: `mark the mode from %v to ERR`

This would provide a little bit more info

Comment (derekbit) [reply pkg/spdk/engine.go]: Updated. Thank you.

**Review by shuo-wu — APPROVED** [review]

**Comment by innobead** [comment]
Well done. LGTM.

**Pull request merged**